src/vs/workbench/browser/parts/views/treeView.ts

```diff
@@ -1183,3 +1214,30 @@ export class CustomTreeView extends TreeView {
 		}
 	}
 }
+
+export class CustomTreeViewDragAndDrop implements ITreeDragAndDrop<ITreeItem> {
+	constructor(private dataSource: TreeDataSource) { }
+
+	onDragOver(data: IDragAndDropData, targetElement: ITreeItem, targetIndex: number, originalEvent: DragEvent): boolean | ITreeDragOverReaction {
+		return { accept: true, bubble: TreeDragOverBubble.Down, autoExpand: true };
+	}
+
+	getDragURI(element: ITreeItem): string | null {
+		return element.tooltip ? element.tooltip.toString() : null;
```

Comment:
This should probably return `URI.revive(element.resource)` if it exists, otherwise, null;

Reply:
The drag and drop does not work if getDragUri returns null. 
Since we are using the element's resourceUri (it's an optional property), should I update the note in the vscode.proposed api or do I return the element handle in case it's null?

Reply:
Returning the handle when there's no `resourceUri` makes sense to me.